### PR TITLE
reapr: fix shasum for samtools dependency

### PR DESCRIPTION
**Pull request by cmdcolin**
Hi there, I found a shasum error in the samtools package installing on linuxbrew. The mac install wasn't affected since there were precompiled bottles

After updating the shasum, the install works fine, and I bumped the revision field also

Let me know if there are any issues

### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [ ] Checked that the tests still pass?

### CODE REVIEW

**Comment by cmdcolin** [comment]
seems to have some trouble building on linuxbrew still perhaps.


build log on linuxbrew


```
brew install reapr
==> Installing reapr from homebrew/science
==> Installing dependencies for homebrew/science/reapr: bamtools
==> Installing homebrew/science/reapr dependency: bamtools
==> Downloading https://linuxbrew.bintray.com/bottles-science/bamtools-2.4.1.x86_64_linux.bottle.tar.gz
Already downloaded: /home/dieshcol/.cache/Homebrew/bamtools-2.4.1.x86_64_linux.bottle.tar.gz
==> Pouring bamtools-2.4.1.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/bamtools/2.4.1: 29 files, 2.3MB
==> Installing homebrew/science/reapr
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Downloading ftp://ftp.sanger.ac.uk/pub/resources/software/reapr/Reapr_1.0.18.tar.gz
Already downloaded: /home/dieshcol/.cache/Homebrew/reapr-1.0.18.tar.gz
==> Downloading http://search.cpan.org/CPAN/authors/id/R/RM/RMBARKER/File-Copy-Link-0.140.tar.gz
Already downloaded: /home/dieshcol/.cache/Homebrew/reapr--File::Spec::Link-0.140.tar.gz
==> perl Makefile.PL INSTALL_BASE=/home/linuxbrew/.linuxbrew/Cellar/reapr/1.0.18_3/libexec
==> make
==> make install
==> Downloading https://github.com/samtools/samtools/commit/64275b4.patch
Already downloaded: /home/dieshcol/.cache/Homebrew/reapr--samtools_patch-4.patch
==> patch -p1 -i /tmp/reapr--samtools_patch-20170621-26695-hv3waj/64275b4.patch -d /tmp/reapr-20170621-26695-ci2gq9/Reapr_1.0.18/third_party/samtools-0.1.18
==> make -C third_party/samtools
==> make -C third_party/samtools razip
==> make -C third_party/tabix
==> make -C third_party/snpomatic
==> make -C src CFLAGS=-I/home/linuxbrew/.linuxbrew/opt/bamtools/include/bamtools
Last 15 lines from /home/dieshcol/.cache/Homebrew/Logs/reapr/09.make:
g++ -I/home/linuxbrew/.linuxbrew/opt/bamtools/include/bamtools -c task_stats.cpp
bam2perfect.o: In function `main':
bam2perfect.cpp:(.text+0xa3): undefined reference to `BamTools::SamHeader::SamHeader(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
bam2perfect.cpp:(.text+0x174): undefined reference to `BamTools::BamReader::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
bam2perfect.cpp:(.text+0x24c): undefined reference to `BamTools::BamWriter::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, BamTools::SamHeader const&, std::vector<BamTools::RefData, std::allocator<BamTools::RefData> > const&)'
bam2perfect.cpp:(.text+0x270): undefined reference to `BamTools::BamWriter::Open(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, BamTools::SamHeader const&, std::vector<BamTools::RefData, std::allocator<BamTools::RefData> > const&)'
bam2perfect.o: In function `bool BamTools::BamAlignment::GetTag<unsigned int>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int&) const':
bam2perfect.cpp:(.text._ZNK8BamTools12BamAlignment6GetTagIjEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERT_[_ZNK8BamTools12BamAlignment6GetTagIjEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERT_]+0xcb): undefined reference to `BamTools::BamAlignment::FindTag(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, char*&, unsigned int const&, unsigned int&) const'
bam2perfect.cpp:(.text._ZNK8BamTools12BamAlignment6GetTagIjEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERT_[_ZNK8BamTools12BamAlignment6GetTagIjEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERT_]+0x1bb): undefined reference to `BamTools::BamAlignment::SetErrorString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
bam2perfect.cpp:(.text._ZNK8BamTools12BamAlignment6GetTagIjEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERT_[_ZNK8BamTools12BamAlignment6GetTagIjEEbRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERT_]+0x291): undefined reference to `BamTools::BamAlignment::SetErrorString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
collect2: error: ld returned 1 exit status
Makefile:71: recipe for target 'bam2perfect' failed
make: *** [bam2perfect] Error 1
make: *** Waiting for unfinished jobs....
make: Leaving directory '/tmp/reapr-20170621-26695-ci2gq9/Reapr_1.0.18/src'

READ THIS: https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.

These open issues may also help:
reapr: fix shasum for samtools dependency https://github.com/Homebrew/homebrew-science/pull/5853
```

**Comment by iMichka** [comment]
There were no Linux bottles here before; so no need to fix the Linux build (of course we are always looking for volunteers :)).

Let's focus on the mac build. If the sha sum has changed; this means that the patch has changed.
Ideally I would like to know what changed. To prevent this from happening in the future, could you please submit the patch here: https://github.com/Homebrew/formula-patches
And then you can directly use it (it does not need to be merged) in the formula. There are multiple examples using formula-patches in this repo. Let me know if you need any help.

**Comment by ilovezfs** [comment]
@iMichka this affects a whole category of patches https://github.com/Homebrew/homebrew-core/pull/13792

The patch has not actually changed.

```
Josephs-MacBook-Pro:~ joe$ new_checksum https://github.com/samtools/samtoolsmit/64275b4.patch
SHA256(/dev/fd/63)= 91ad8e714c0d48a7b3dde99eb4eba49c00b8e88f0c4b67cdd284c208a4741803
Josephs-MacBook-Pro:~ joe$ old_checksum https://github.com/samtools/samtools/commit/64275b4.patch
SHA256(/dev/fd/63)= daa4d7050b56e089e3016aeb8fec9187c7e8f3bc1164d4d0c8aab1f61efe8b89
Josephs-MacBook-Pro:~ joe$ 
```

```
old_checksum(){(openssl dgst -sha256 <(curl -sL "$1" | sed -E 's/(index .......).*(\.\........).*( .*)/\1\2\3/'))}
new_checksum(){(openssl dgst -sha256 <(curl -sL "$1"))}
```

File: reapr.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class Reapr < Formula
   # tag "bioinformatics"
   url "ftp://ftp.sanger.ac.uk/pub/resources/software/reapr/Reapr_1.0.18.tar.gz"
   sha256 "6d691b5b49c58aef332e771d339e32097a7696e9c68bd8f16808b46d648b6660"
-  revision 2
+  revision 3
```
Comment (ilovezfs) [inline reapr.rb]: The revision does not need to be bumped.

**Comment by cmdcolin** [comment]
Gotcha, removed that commit

**Comment by cmdcolin** [comment]
Good catch on the patch format changing...

**Comment by cmdcolin** [comment]
So basically since shasum issue figured out, the test does still fail due to linuxbrew build


I guess that might need fixing although it was likely broken before this PR anyways

**Comment by ilovezfs** [comment]
@cmdcolin no existing Linux bottle so :man_shrugging:

**Comment by ilovezfs** [comment]
Thanks for the PR! :rocket:

**Pull request merged**